File: leetcode/_contest/bi-weekly-contest-14/1.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func toHexspeak(num string) string {
	n, _ := strconv.Atoi(num)
	s := strings.ToUpper(fmt.Sprintf("%0x", n))
	res := ""

	for _, ch := range s {
		if ch == '1' {
			res += "I"
			continue
		}

		if ch == '0' {
			res += "O"
			continue
		}

		if ch < 'A' {
			return "ERROR"
		}

		res += string(ch)
	}
	return res
}

func main() {
	fmt.Println(toHexspeak("257"))
	fmt.Println(toHexspeak("747823223228"))
}

File: leetcode/_contest/bi-weekly-contest-14/1.js
/**
 * @param {string} num
 * @return {string}
 */
var toHexspeak = function(num) {
  const s = parseInt(num).toString(16).toUpperCase()
  let res = ''

  for (const ch of s) {
    if (ch === '1') {
      res += 'I'
      continue
    }
    if (ch === '0') {
      res += 'O'
      continue
    }

    if ('ABCDEFIO'.indexOf(ch) < 0) {
      res = 'ERROR'
      break
    }
    res += ch
  }
  return res
}

//console.log(toHexspeak(257))
//console.log(toHexspeak(3))
console.log(toHexspeak('747823223228'))
